deploy/cluster_role.yaml

```diff
@@ -36,9 +36,10 @@ rules:
     resources:
       - consolelinks
     verbs:
-      - create
       - get
-      - delete
       - list
-      - patch
+      - watch
```

Comment:
You should be able to skip the `watch` right by using the new [`nonCachedClient`](https://github.com/eclipse/che-operator/blob/9682f3448fe240c216aa22d9d3f56cc056b01494/pkg/controller/che/che_controller.go#L204) reconciler attribute.

It allows accessing the K8S API without trying to further cache / watch the related K8S objects, which doesn't seem to be necessary here.

Reply:
Current solution tries to get and update Che Cluster each time when che operator is invoked.
So, it makes sense to have `watch` here to avoid retrieving it each time
![Screenshot_20190903_115130](https://user-images.githubusercontent.com/5887312/64176308-d6aa1400-ce65-11e9-83c5-d963e73494c6.png)

Or maybe you would suggest another solution, like:
1. Do not update console link but only create it when it does not exist. But I see an error here, since existing consolelink may be outdated.
2. something else.

Thanks.

Reply:
Afaik the Che operator is not called if no change occurred in the `CheCluster` resource.
That's not going to happen that regularly, and I don't see any value / gain trgigerring regular watch of a `ConsoleLink` object. The only impact is increasing the range of required permissions.

Reply:
OK, will get rid of watch verb here at least for the time being.

Reply:
done